.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,67 @@
+name: deploy
+
+on:
+  push:
+    branches:
+    - main
+  release:
+    types: [published]
+
+env:
+  REGISTRY: quay.io
+  IMAGE_NAME: operator-framework/plain-v0-provisioner
+
+jobs: 
+  main:
+    if: ${{ github.event_name == 'push' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Login to quay.io
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ secrets.QUAY_USER }}
+          password: ${{ secrets.QUAY_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+          context: .
+          push: true
+          tags: main
+          labels: ${{ steps.meta.outputs.labels }}
+
+  release:
```

Comment:
In the past, I've never had to duplicate the jobs in their entirety. We're doing the exact same steps in both, just with different tags right?

Seems like it should be possible to set the desired push tag by using [type=ref](https://github.com/docker/metadata-action#typeref), which does exactly what we want for both branches and tags.

Reply:
Yep - another thing Tyler/I were talking about offline. He's looking into consolidating this.